### PR TITLE
Rewrite PackageCollectionsSigning using swift-certificates

### PR DESCRIPTION
**Pull request by yim-lee**
Motivation:
`PackageCollectionsSigning` was written before `swift-certificates` and `swift-asn1` were available. The implementation relied on BoringSSL C library via `CCryptoBoringSSL`, which is not meant to be used outside of `swift-crypto` because it introduces Swift-breaking changes often (as SwiftPM had experienced with version `2.4.1`).

Modifications:
- Rewrite `PackageCollectionsSigning` using `swift-certificates` and `swift-asn1` so we can remove `CCryptoBoringSSL` dependency.
- `PackageCollectionsSigningLibC`, which contains SwiftPM's implementation of OCSP, is no longer needed and removed in this PR.

### CODE REVIEW

File: Sources/PackageCollectionsSigning/CertificatePolicy.swift [hunk 1]
```diff
@@ -0,0 +1,644 @@
...
+import Basics
+@_implementationOnly import SwiftASN1
+@_implementationOnly import X509
+
+public enum CertificatePolicyKey: Hashable, CustomStringConvertible {
+    case `default`(subjectUserID: String? = nil, subjectOrganizationalUnit: String? = nil)
+    case appleSwiftPackageCollection(subjectUserID: String? = nil, subjectOrganizationalUnit: String? = nil)
+
+    @available(*, deprecated, message: "use `appleSwiftPackageCollection` instead")
+    case appleDistribution(subjectUserID: String? = nil, subjectOrganizationalUnit: String? = nil)
```
Comment (yim-lee) [inline Sources/PackageCollectionsSigning/CertificatePolicy.swift:25]: Deprecated

File: Sources/PackageCollectionsSigning/CertificatePolicy.swift [hunk 2]
```diff
@@ -0,0 +1,644 @@
...
+        callback: @escaping (Result<Void, Error>) -> Void
+    ) {
+        let wrappedCallback: (Result<Void, Error>) -> Void = { result in callbackQueue.async { callback(result) } }
+
+        guard !certChain.isEmpty else {
+            return wrappedCallback(.failure(CertificatePolicyError.emptyCertChain))
+        }
+
+        Task {
+            var trustStore = CertificateStores.defaultTrustRoots
```
Comment (yim-lee) [inline Sources/PackageCollectionsSigning/CertificatePolicy.swift:91]: A major change in the new implementation is that we no longer use the system trust store (we used to on Darwin platforms through the `Security` framework) and SwiftPM will its own default trust store instead.

File: Sources/PackageCollectionsSigning/embedded_resources.swift
Comment (yim-lee) [inline Sources/PackageCollectionsSigning/embedded_resources.swift:1]: Both this and `PackageSigning` require the same resources. Would it make sense to have a shared resource-only module?

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Review by tomerd — APPROVED** [review]

File: Package.swift [hunk 3]
```diff
@@ -66,44 +66,6 @@ automatic linking type with `-auto` suffix appended to product's name.
...
-    "PackageCollectionsModel",
-]
-// swift-crypto's Crypto module depends on CCryptoBoringSSL on these platforms only
-#if os(Linux) || os(Windows) || os(Android)
-packageCollectionsSigningTargets.append(
-    .target(
-        /** Package collections signing C lib */
-        name: "PackageCollectionsSigningLibc",
-        dependencies: [
-            .product(name: "Crypto", package: "swift-crypto"), // for CCryptoBoringSSL
```
Comment (tomerd) [inline Package.swift:82]: done still need the dependency on swift-crypto?

Comment (yim-lee) [reply Package.swift:82]: Yes, we still need it for keys and signing.

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Pull request merged**